lib/Runtime/Library/ConcatString.cpp

```diff
@@ -130,15 +130,27 @@ namespace Js
         scriptContext->GetOrAddPropertyRecord(this->GetSz(), static_cast<int>(this->GetLength()),
             (Js::PropertyRecord const **)&propertyRecord);
 
-        this->propertyString = scriptContext->GetPropertyString(propertyRecord->GetPropertyId());
-        this->SetBuffer(this->propertyString->GetString()); // use the same buffer
```

Comment:
This was the line causing OS14564548. i.e. LiteralStringWith... has buffer from propertyRecord and for some reason we copy the string into another context but keep the buffer.. things are getting interesting.

Reply:
Moved this to #4172